src/CocktailDetails/CocktailDetails.tsx

```diff
@@ -13,13 +13,28 @@ const CocktailDetails: React.FC<CocktailDetailsProps> = (props) => {
 	const getCocktail = async ():Promise<any> => {
 		try {
 			const data: RandomCocktail = await getCocktailDetails(props.id);
-			console.log(data);
-			setCocktailInfo(data);
+			setCocktailInfo(removeNulls(data));
 		} catch (error) {
 		setError(error.message);
 		}
 	}
 
+	const removeNulls = (info: RandomCocktail): RandomCocktail => {
+  	const drinkDetails: Partial<RandomCocktail> = {idDrink: '', strDrink: '', strInstructions: '', strDrinkThumb: ''};
+		(Object.keys(info) as Array<keyof RandomCocktail>).forEach((detail) => {
+      if (info[detail] !== null) { 
+        drinkDetails[detail] = info[detail]; 
+      }
+		});
+    return drinkDetails as RandomCocktail;
+	}
+	
+	const displayIngredients = (endingIndex: number, givenKey: string): string[] => {
+		const cocktailIngredients = (Object.keys(cocktailInfo) as Array<keyof RandomCocktail>)
+			.filter(keys => keys.slice(0, endingIndex) === givenKey);
+		return cocktailIngredients.map(i => cocktailInfo[i] as string);
+  }
```

Comment:
Partial, keyof, and asserting 'as string' were crucial in this block of code for TypeScript. B/c within the scope of a prototype such as .filter, .map, etc., the values become dynamic but TS doesn't like that since there's possibilities for an undefined or null value. So line 35 asserts the values will be strings to strictly type it.

Reply:
Whoa! That's crazy.  Nice work.  I would like to go over this in our stand up, as I'm still a little fuzzy on the logic.

Reply:
Yes of course! I don't fully understand it, Hannah & my 1/2 mentor helped me get this hah. But will tell what I know and understand! @Melizzo

Reply:
We need to give Hannah a proper shout out for this project.  She's been an angel.